pliers/extractors/audio.py

```diff
@@ -488,4 +494,88 @@ class PercussiveExtractor(LibrosaFeatureExtractor):
     For details on argument specification visit:
     https://librosa.github.io/librosa/effect.html.'''
 
+
     _feature = 'percussive'
+
+
+class AudiosetLabelExtractor(AudioExtractor):
+
+    ''' Extract probability of audio event classes based on AudioSet-YouTube 
+    corpus using a YAMNet architecture. Code available at:
+    https://github.com/tensorflow/models/tree/master/research/audioset/yamnet 
+
+    Args:
+    hop_size (float): size of the window (in seconds) on which label extraction
+        is performed.
+    top_n (int): specifies how many of the highest label probabilities are 
+        returned. If not defined, returns probabilities for all labels.
+    spectrogram (bool): if True, plots mel-spectrogram of the audio stimulus.
+    yamnet_kwargs (optional): Optional named arguments that modify input 
+        parameters for the model (see params.py file in yamnet repository)
+    '''
+
+    _log_attributes = ('params', 'top_n')
+
+    def __init__(self, hop_size=0.1, top_n=None, 
+                 spectrogram=False, **yamnet_kwargs):
+        verify_dependencies(['yamnet'])
+        verify_dependencies(['tensorflow'])
+
+        MODULE_PATH = path.dirname(yamnet.__file__)
+        WEIGHTS_PATH = path.join(MODULE_PATH, 'yamnet.h5')
+        LABELS_PATH = path.join(MODULE_PATH, 'yamnet_class_map.csv')
+
+        self.spectrogram = spectrogram
+        self.params = yamnet.params
```

Comment:
Maybe only the kwarg params should be logged. That is, only log if you changed from the defaults

Reply:
edited log_attributes as suggested